### PR TITLE
GetCloudAppProperties

### PR DESCRIPTION
**Pull request by JackLivio**
This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Summary
Implements new rpc to be used by oem cloud app stores. Retrieves information related to cloud apps when given the cloud apps app_id.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/interfaces/MOBILE_API.xml [hunk 1]
```diff
@@ -6678,6 +6697,40 @@
...
+    <function name="GetCloudAppProperties" functionID="GetCloudAppPropertiesID" messagetype="request" since="5.1">
+        <description>
+            RPC used to get the current properties of a cloud application
+        </description> 
+        <param name="appID" type="String" maxlength="100" mandatory="true"></param>
+    </function>
+
+    <function name="GetCloudAppProperties" functionID="GetCloudAppPropertiesID" messagetype="response" since="5.1">
+        <description>The response to GetCloudAppProperties</description>
+        <param name="properties" type="CloudAppProperties" mandatory="false">
```
[the remark this answers is not in the document]
Comment (JackLivio) [reply src/components/interfaces/MOBILE_API.xml]: Yes

File: src/components/interfaces/MOBILE_API.xml [hunk 2]
```diff
@@ -6638,19 +6657,8 @@
...
-        </param>
-        <param name="cloudAppAuthToken" type="String" maxlength="100" mandatory="false">
-            <description>Used to authenticate websocket connection on app activation</description>
-        </param>
-        <param name="cloudTransportType" type="String" maxlength="100" mandatory="false">
-            <description>Specifies the connection type Core should use</description>
-        </param>
-        <param name="hybridAppPreference" type="HybridAppPreference" mandatory="false">
-            <description>Specifies the user preference to use the cloud app version or mobile app version when both are available</description>
+        <param name="properties" type="CloudAppProperties" mandatory="false">
```
Comment (jacobkeeler) [inline src/components/interfaces/MOBILE_API.xml]: Should be mandatory

File: src/components/interfaces/MOBILE_API.xml [hunk 3]
```diff
@@ -552,6 +552,24 @@
...
+        <param name="authToken" type="String" maxlength="65535" mandatory="false">
+            <description>Used to authenticate websocket connection on app activation</description>
+        </param>
+        <param name="cloudTransportType" type="String" maxlength="100" mandatory="false">
+            <description>Specifies the connection type Core should use</description>
+        </param>
+        <param name="hybridAppPreference" type="HybridAppPreference" mandatory="false">
+            <description>Specifies the user preference to use the cloud app version or mobile app version when both are available</description>
+        </param>
+        <param name="endpoint" type="String" maxlength="65535" mandatory="false"></param>
```
Comment (jacobkeeler) [inline src/components/interfaces/MOBILE_API.xml]: You should be able to set this field or we need to include a note that this cannot be sent in a `Set` request

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_cloud_app_properties_request.cc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  SendResponse(true, mobile_apis::Result::SUCCESS, NULL, &response_params);
+}
+
+void GetCloudAppPropertiesRequest::on_event(
+    const app_mngr::event_engine::Event& event) {
+  LOG4CXX_INFO(logger_, "GetCloudAppPropertiesRequest on_event");
+}
+
+}  // namespace commands
+}  // namespace sdl_rpc_plugin
```
Comment (jacobkeeler) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_cloud_app_properties_request.cc]: Please add newlines to the end of new files

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_cloud_app_properties_request.cc [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  LOG4CXX_AUTO_TRACE(logger_);
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "Application is not registered");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  if ((*message_)[strings::msg_params].empty()) {
```
Comment (jacobkeeler) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_cloud_app_properties_request.cc]: Shouldn't need this case, `app_id` is mandatory

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_cloud_app_properties_request.cc [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                                        enabled,
+                                        endpoint,
+                                        certificate,
+                                        auth_token,
+                                        cloud_transport_type,
+                                        hybrid_app_preference);
+
+  ApplicationSharedPtr cloud_app =
+      application_manager_.application_by_policy_id(policy_app_id);
+  std::string cloud_name = "";
```
Comment (jacobkeeler) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_cloud_app_properties_request.cc]: `= ""` is redundant

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_cloud_app_properties_request.cc [hunk 7]
```diff
@@ -35,7 +35,9 @@ void SetCloudAppPropertiesRequest::Run() {
     return;
   }
 
-  policy_handler_.OnSetCloudAppProperties(*message_);
+  if ((*message_)[strings::msg_params].keyExists(strings::properties)) {
```
Comment (jacobkeeler) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/set_cloud_app_properties_request.cc]: Properties is mandatory, shouldn't need this case

**Pull request merged**